internal/conditions/conditions.go

```diff
@@ -6,82 +6,73 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-func newCondition(typeVal, reason, message string, status metav1.ConditionStatus) metav1.Condition {
+func NotRequired(object string) metav1.Condition {
 	return metav1.Condition{
-		Type:    typeVal,
-		Status:  status,
-		Reason:  reason,
-		Message: message,
+		Type:    fmt.Sprintf("%sReconciled", object),
+		Reason:  fmt.Sprintf("%sNotRequired", object),
+		Message: fmt.Sprintf("%s is not required", object),
+		Status:  metav1.ConditionTrue,
```

Comment:
While I was debugging conditions, I realized that the `newCondition` function wasn't earning us much, so I just inlined the conditions below to make it more clear what is used for which value.

Reply:
Yeah I wavered on that change, works for me 👍🏼